### ISSUE TITLE
PropsSI array argument functionality gives unexpected results for arrays of length 1 or 0

### ISSUE THREAD

**Issue by danielkwalsh** [CoolProp/CoolProp#2417]
### Description

When using PropsSI with lists as arguments, I get unexpected results for lists of length 0 or 1.

### Steps to Reproduce
**Case 1: Length 1 inputs:**

```
import CoolProp.CoolProp as CP
result = CP.PropsSI('T', 'P', [101325], 'Q', [0], 'Ammonia') #Values don't matter, just array length
len(result) #gives error: TypeError: len() of unsized object
```

**Case 2: Length 0 inputs:**
```
import CoolProp.CoolProp as CP
result = CP.PropsSI('T', 'P', [], 'Q', [], 'Ammonia') #Now length-zero arrays
print(result) #gives error: TypeError: len() of unsized object
```

**Expected behavior:**
Both of the above should work and return a length-1 and length-0 array, respectively

**Actual behavior:** [What actually happens]
Length-1 arrays return "unsized objects", and length-0 give an error.
### Versions

**CoolProp Version:** 6.6.0
**Operating System and Version:** 14.6.1 (23G93)  
**Access Method:** Python (ipynb)

### Additional Information

If possible, please post examples and/or screenshots of the issue.

**Comment by ibell** [comment]
Do you get the same result with one-element numpy array?

**Comment by danielkwalsh** [comment]
@ibell, yes, both cases behave the same as before even when using numpy arrays.

**Comment by danielkwalsh** [comment]
@ibell, I know I can import numpy in this environment, since you asked me to try running it with numpy arrays, which I did by importing numpy to do so.

`import CoolProp.CoolProp as CP
import numpy as np
result = CP.PropsSI('T','P',np.array([101325]),'Q', np.array([0]),'Ammonia')
len(result)`

Python version is 3.11.5.
